### ISSUE TITLE
Documentation mistake of Adam in v1.6.0?

### ISSUE THREAD

**Issue by yuhaozhang** [pytorch/pytorch#42843]
## 📚 Documentation

The documentation of Adam since v1.6.0 suggests that it uses the weight decay fix proposed by paper "Decoupled Weight Decay Regularization":
https://github.com/pytorch/pytorch/blob/4b4273a04e566dcdf4cb96882f49c41127e8d3f1/torch/optim/adam.py#L10-L11

However I found this to not be the case. The actual weight decay in Adam is still the old one:
https://github.com/pytorch/pytorch/blob/4b4273a04e566dcdf4cb96882f49c41127e8d3f1/torch/optim/adam.py#L99-L100

In contrast, the new weight decay fix is implemented in AdamW, and explained by the AdamW doc:
https://github.com/pytorch/pytorch/blob/4b4273a04e566dcdf4cb96882f49c41127e8d3f1/torch/optim/adamw.py#L10

https://github.com/pytorch/pytorch/blob/4b4273a04e566dcdf4cb96882f49c41127e8d3f1/torch/optim/adamw.py#L73

It is likely that I misunderstood the current documentation of Adam, but I found it confusing and suggest that there is no difference between the Adam and AdamW implementation, while the difference still exists.

**Comment by zou3519** [comment]
cc @vincentqb -- what do you think?

**Comment by vincentqb** [comment]
The reference to the follow-up paper was added via [this issue](https://github.com/pytorch/pytorch/issues/41477#issue-657482282), and the justification for the change is [here](https://github.com/pytorch/pytorch/issues/41477#issue-657482282). Does this address your comment? If not, how would you suggest rephrasing the documentation?

**Comment by yuhaozhang** [comment]
Hi @vincentqb, thanks for the followup! Your understanding is correct in that this current reference leads to more confusion about the actual weight decay applied in the Adam implementation.

I have looked at https://github.com/pytorch/pytorch/issues/41477 as you pointed to, and I am now more confused about why this issue was brought up in the first place. It looks to me that the current Adam implementation conforms perfectly to the [original Adam paper](https://arxiv.org/pdf/1412.6980.pdf) (Algorithm 1 in page 2), and there is no difference between how epsilon was applied in the original Adam paper and the more recent [Decoupled Weight Decay paper](https://arxiv.org/pdf/1711.05101.pdf). 

Looking further down the line, it looks like there was originally a difference between how Adam was implemented in early versions of PyTorch, which was then fixed (https://github.com/pytorch/pytorch/commit/fed5ca192c98619947cca2fb9491ac9624b787d2) by referencing to the more recent Decoupled Weight Decay paper. However this reference is not necessary since the implementation of epsilon is the same in both papers and we can just equally reference the original Adam paper for that.

**Comment by yuhaozhang** [comment]
To summarize, a proposed change would be to remove this note at all, because: 1) the current code implements epsilon in the same way as the original Adam paper; 2) adding this note may lead to further confusion about the weight decay in `Adam`. There is a chance that my understanding of this is wrong, in which case please correct me. Thanks!

**Comment by dvolgyes** [comment]
I absolutely  think this documentation mistake is serious, and the documentation should be fixed as soon as possible.
It mislead at least me, but probably other people too.

**Comment by Yura52** [comment]
+1, I have always thought that the documentation says that Adam and AdamW are literally the same (which was strange to me at the same time).
I think the remark can be safely removed, in its current form it rather adds confusion than makes things clearer. If it is important to keep it, then I would propose to add a note, something like this:  
"**Note**: for the decoupled weight decay, use AdamW."
